packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,6 +336,53 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
 
     private createTiptapEditor(): Editor {
         const customLink = this.getCustomLinkExtension();
+        const validAbsoluteLinkRegex = /^https?:\/\//i;
+
+        /**
+         * @param htmlString contains the html string of the copied content. If the content is a link, the `htmlString` contains anchor tag and a href value.
+         * ProseMirror reference for `transformPastedHTML`: https://prosemirror.net/docs/ref/#view.EditorProps.transformPastedHTML
+         */
+        const transformPastedHTML = (htmlString: string): string => {
+            const templateElement = document.createElement('template');
+            templateElement.innerHTML = htmlString;
+            const templateDocument = templateElement.content.ownerDocument;
+
+            templateElement.content
```

Comment:
I'd personally be surprised if a hyperlink (text backed by a link) was replaced by the URL on paste. I would expect the text contents to be pasted and the link to be stripped away. I'm basing this on the behavior I see in plain text editors like VSCode.

What's the basis for the currently implemented behavior? Is there a PO request or customer workflow that requires it?

Reply:
As we mentioned in the other group, we quickly aligned with Josh and it is also a request from him to paste it as a link when copied from the Edge browser. 

Due to the limitations of supporting only the absolute link without the hyperlink, we could come up with this approach when pasting only the URL when the hyperlinks are copied and pasted. Since this is impacting directly the user's point of view who might be using Edge browsers they struggle to identify what is pasted here in the editor.

We are currently capturing all the hyperlink workaround removal in a separate issue. We will remove this workaround when the support is added as we aligned before.

Reply:
This behavior was implemented for [Bug 2525823](https://dev.azure.com/ni/DevCentral/_workitems/edit/2525823): Rich text editor | Copying a link from Edge browser which is not pasting as a link

### **Edge browser**
1. When copying URL from Edge browser's URL bar, we have both types `text/plain` and `text/html` available. 
![image](https://github.com/ni/nimble/assets/94439533/917323fd-f9a6-49de-9e9b-f0a3fc339a56)

2. If we look at the options in edge browser for pasting - link paste seems to be default. Below is a snip of edge browser with Google search loaded and I was trying to copy and paste the link from URL bar
![image](https://github.com/ni/nimble/assets/94439533/759a1289-3226-4977-ad8a-585ba6970982)


### **Chrome & Firefox browser**
In chrome and firefox, while copying, it copies only `text/plain` which leaves us to paste only plain text
![image](https://github.com/ni/nimble/assets/94439533/7f803d37-dfee-4f56-bf18-d6f6dd7a0c20)

### **Copying nimble anchor**
However copying `nimble-anchor` is adding additional inline styles which we might need to investigate further through a separate issue for `nimble-anchor`
![image](https://github.com/ni/nimble/assets/94439533/11a5bd8e-fdea-4a13-9c82-51de52d3b807)